### PR TITLE
Private endpoints and JEDI controllers

### PR DESCRIPTION
**Pull request by pajind**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [x] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by pajind** [comment]
The changes are according to the apis which are already open to the public on the defined api version. The earlier swagger was wrong and had incorrect details. We will make sure the swagger is not published until we have finalized on apis completely.

@jianyexi  : Please review the given PR. The earlier swagger was incorrect and the new apis are the right apis which are open to public

**Comment by pilor** [comment]
Please complete the checklist in the PR description and fix the lintdiff/modelvalidation failures

**Comment by pilor** [comment]
Please also follow the breaking change approval process outlined in this comment:  https://github.com/Azure/azure-rest-api-specs/pull/14776#issuecomment-864526246

**Comment by pajind** [comment]
. 

> Please also follow the breaking change approval process outlined in this comment: [#14776 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/14776#issuecomment-864526246)

#pilor : Have raised a request for a breaking change approval : Scenario 10264488: [Azure Migrate]: Added new APIs to existing API version

**Comment by pilor** [comment]
@jorgecotillo is the API reviewer this week and can take a look at this

**Comment by pajind** [comment]
@pilor , @@jorgecotillo  please review the PR. I have fixed all Lintdiff errors and the remaining errors cant be fixed due to breaking changes in existing apis. Please review it once so I will schedule a meeting for the comments and unresolved errors to discuss

**Comment by pilor** [comment]
> @pilor , @jorgecotillo please review the PR. I have fixed all Lintdiff errors and the remaining errors cant be fixed due to breaking changes in existing apis. Please review it once so I will schedule a meeting for the comments and unresolved errors to discuss

Please resolve merge conflcts.  A review didn't start on this PR yet so the current week's reviewer is who should pick it up (@majastrz )

File: specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json [hunk 1]
```diff
@@ -2968,16 +3450,226 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Projects_List": {
+            "$ref": "./examples/VMwareSites_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.OffAzure/HyperVSites": {
```
Comment (majastrz) [inline specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:3572]: >HyperVSites [](http://example.com/codeflow?start=66&length=11)

You should also have a GET operation for a single resource of type `Microsoft.OffAzure/HyperVSites`.

File: specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json [hunk 2]
```diff
@@ -2968,16 +3450,226 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Projects_List": {
+            "$ref": "./examples/VMwareSitesInSubscription_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OffAzure/VMwareSites": {
```
Comment (majastrz) [inline specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:3518]: >VMwareSites [](http://example.com/codeflow?start=101&length=11)

You should also have a GET operation for a single resource of type `Microsoft.OffAzure/VMwareSites`

File: specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json (hunk 1 above)
Comment (pajind) [reply specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:3572]: Its there "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OffAzure/HyperVSites/{siteName}", line 918

File: specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json (hunk 2 above)
Comment (pajind) [reply specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:3518]: Its there, "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OffAzure/VMwareSites/{siteName}", line 1836

Comment (majastrz) [reply specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:3570]: I missed it before. Thanks for pointing it out!

File: specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json (hunk 1 above)
Comment (majastrz) [reply specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:3628]: I missed it before. Thanks for pointing it out!

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by pajind** [comment]
@JianyeXi Can you please check and merge the PR. I got ARM sign off

**Comment by pajind** [comment]
@jianyexi Can you please check and merge. Its a blocker for our JEDI deadlines

**Comment by jianyexi** [comment]
@pajind  could you fix the lint errors https://github.com/Azure/azure-rest-api-specs/pull/14776/checks?check_run_id=3542878612 ?
Why the model or property name start with capital character

**Comment by pajind** [comment]
> @pajind could you fix the lint errors https://github.com/Azure/azure-rest-api-specs/pull/14776/checks?check_run_id=3542878612 ?
> Why the model or property name start with capital character

These Lintdiff errors are for existing apis and were not there in earlier reviews. The service is years old. I can take all these errors/breaking changes including camelcase in next api version while increasing api version

**Comment by jianyexi** [comment]
> > @pajind
> >             
> >             Pallavi Jindal
> >             FTE could you fix the lint errors https://github.com/Azure/azure-rest-api-specs/pull/14776/checks?check_run_id=3542878612 ?
> > Why the model or property name start with capital character
> 
> @jianyexi Jianye Xi FTE These Lintdiff errors are for existing apis and were not there in earlier reviews. The service is years old. I can take all these errors/breaking changes including camelcase in next api version while increasing api version. This api version is already public

Okay, I approved it ,as it's not newly introduced errors.

File: specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json [hunk 3]
```diff
@@ -2799,16 +2938,19 @@
           }
         ],
         "responses": {
-          "200": {
-            "description": "OK",
-            "schema": {
-              "$ref": "#/definitions/MasterSite"
+          "202": {
+            "description": "Accepted",
```
Comment (mikhailshilkov) [inline specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-07-07/migrate.json:2942]: This looks wrong to me. Now, there's no way to tell what the shape of the eventual response is going to be. I believe all other resources declare their final response, even if in practice it's a long running operation, similarly to what the previous version did.

**Pull request merged**